File: RESTFul/AAB/PHP/app1/src/aabUtil.php
<?php
/*
 * Copyright 2015 AT&T
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

require_once __DIR__ . '/../lib/Util/Util.php';

use Att\Api\Util\Util;

function generatePhonesTable($contactId, $phones) {
    if ($phones === null || count($phones) == 0)
        return null;

    $table = array();
    $table['caption'] = "Contact ($contactId) Phones:";
    $table['headers'] = array('type', 'number', 'preferred');
    $values = array();
    foreach ($phones as $phone) {
        $values[] = Util::convertNulls(array(
            $phone->getPhoneType(),
            $phone->getNumber(),
            $phone->isPreferred(),
        ));
    }
    $table['values'] = $values;
    return $table;
}

function generateEmailsTable($contactId, $emails) {
    if ($emails === null || count($emails) == 0)
        return null;

    $table = array();
    $table['caption'] = "Contact ($contactId) Emails:";
    $table['headers'] = array('type', 'address', 'preferred');
    $values = array();
    foreach ($emails as $email) {
        $values[] = Util::convertNulls(array(
            $email->getEmailType(),
            $email->getEmailAddress(),
            $email->isPreferred(),
        ));
    }
    $table['values'] = $values;
    return $table;
}

function generateImsTable($contactId, $ims) {
    if ($ims === null || count($ims) == 0)
        return null;

    $table = array();
    $table['caption'] = "Contact ($contactId) Ims:";
    $table['headers'] = array('type', 'uri', 'preferred');
    $values = array();
    foreach ($ims as $im) {
        $values[] = Util::convertNulls(array(
            $im->getImType(),
            $im->getImUri(),
            $im->isPreferred(),
        ));
    }
    $table['values'] = $values;
    return $table;
}

function generateAddressesTable($contactId, $addresses) {
    if ($addresses === null || count($addresses) == 0)
        return null;

    $table = array();
    $table['caption'] = "Contact ($contactId) Addresses:";
    $table['headers'] = array(
        'type', 'preferred', 'poBox', 'addressLine1', 'addressLine2',
        'city', 'state', 'zipcode', 'country'
    );
    $values = array();
    foreach ($addresses as $address) {
        $values[] = Util::convertNulls(array(
            $address->getAddressType(),
            $address->isPreferred(),
            $address->getPoBox(),
            $address->getAddressLineOne(),
            $address->getAddressLineTwo(),
            $address->getCity(),
            $address->getState(),
            $address->getZipCode(),
            $address->getCountry(),
        ));
    }
    $table['values'] = $values;
    return $table;
}

function generateWeburlsTable($contactId, $weburls) {
    if ($weburls === null || count($weburls) == 0)
        return null;

    $table = array();
    $table['caption'] = "Contact ($contactId) Weburls:";
    $table['headers'] = array('type', 'url', 'preferred');
    $values = array();
    foreach ($weburls as $weburl) {
        $values[] = Util::convertNulls(array(
            $weburl->getWebUrlType(),
            $weburl->getUrl(),
            $weburl->isPreferred(),
        ));
    }
    $table['values'] = $values;
    return $table;
}

function generateQuickContactTable($qcontact) {
    $table = array();
    $table['caption'] = 'Quick Contact:';
    $table['headers'] = array(
        'contactId', 'formattedName', 'firstName', 'middleName', 'lastName',
        'prefix', 'suffix', 'nickName', 'organization',
    );
    $values = array();
    $values[] = array(
        $qcontact->getContactId(), $qcontact->getFormattedName(),
        $qcontact->getFirstName(), $qcontact->getMiddleName(),
        $qcontact->getLastName(), $qcontact->getPrefix(),
        $qcontact->getSuffix(), $qcontact->getNickname(),
        $qcontact->getOrganization(),
    );
    $table['values'] = $values;
    return $table;
}

function generateContactTable($contact) {
    $table = array();
    $table['caption'] = 'Contact:';
    $table['headers'] = array(
        'contactId', 'creationDate', 'modificationDate', 'formattedName',
        'firstName', 'lastName', 'prefix', 'suffix', 'nickName',
        'organization', 'jobTitle', 'anniversary', 'gender', 'spouse',
        'hobby', 'assistant',
    );
    $values = array();
    $values[] = array(
        $contact->getContactId(), $contact->getCreationDate(),
        $contact->getModificationDate(), $contact->getFormattedName(),
        $contact->getFirstName(), $contact->getLastName(),
        $contact->getPrefix(), $contact->getSuffix(),
        $contact->getNickname(), $contact->getOrganization(),
        $contact->getJobTitle(), $contact->getAnniversary(),
        $contact->getGender(), $contact->getSpouse(), $contact->getHobby(),
        $contact->getAssistant(),
    );
    $table['values'] = $values;
    return $table;
}

/* vim: set expandtab tabstop=4 shiftwidth=4 softtabstop=4: */
?>

File: RESTFul/AAB/Java/app1/src/main/java/com/att/api/aab/controller/AabUtil.java
package com.att.api.aab.controller;

import org.json.JSONArray;
import org.json.JSONObject;

import com.att.api.aab.service.Address;
import com.att.api.aab.service.Contact;
import com.att.api.aab.service.Email;
import com.att.api.aab.service.Im;
import com.att.api.aab.service.Phone;
import com.att.api.aab.service.QuickContact;
import com.att.api.aab.service.WebURL;

public final class AabUtil {
    private AabUtil() {
    }

    public static JSONObject generatePhonesTable(
        final String contactId, final Phone[] phones
    ) {
        JSONObject jtable = new JSONObject();
        jtable.put("caption", "Contact (" + contactId + ") Phones:");
        jtable.put("headers", new JSONArray().put("type").put("number").put("preferred"));
        JSONArray jphones = new JSONArray();
        for (final Phone phone : phones) {
            final JSONArray jphone = new JSONArray();
            jphone
                .put(phone.getType())
                .put(phone.getNumber())
                .put(phone.isPreferred());
            jphones.put(jphone);
        }
        jtable.put("values", jphones);
        return jtable;
    }

    public static JSONObject generateEmailsTable(
        final String contactId, final Email[] emails
    ) {
        JSONObject jtable = new JSONObject();
        jtable.put("caption", "Contact (" + contactId + ") Emails:");
        jtable.put("headers", new JSONArray().put("type").put("address").put("preferred"));
        JSONArray jemails = new JSONArray();
        for (final Email email : emails) {
            final JSONArray jemail = new JSONArray();
            jemail
                .put(email.getType())
                .put(email.getEmailAddress())
                .put(email.isPreferred());
            jemails.put(jemail);
        }
        jtable.put("values", jemails);
        return jtable;
    }

    public static JSONObject generateImsTable(
        final String contactId, final Im[] ims
    ) {
        JSONObject jtable = new JSONObject();
        jtable.put("caption", "Contact (" + contactId + ") Ims:");
        jtable.put("headers", new JSONArray().put("type").put("uri").put("preferred"));
        JSONArray jims = new JSONArray();
        for (final Im im : ims) {
            final JSONArray jim = new JSONArray();
            jim
                .put(im.getType())
                .put(im.getImUri())
                .put(im.isPreferred());
            jims.put(jim);
        }
        jtable.put("values", jims);
        return jtable;
    }

    public static JSONObject generateAddressesTable(
        final String contactId, final Address[] addresses
    ) {
        JSONObject jtable = new JSONObject();
        jtable.put("caption", "Contact (" + contactId + ") Addresses:");
        jtable.put("headers", 
            new JSONArray()
                .put("type")
                .put("preferred")
                .put("poBox")
                .put("addressLine1")
                .put("addressLine2")
                .put("city")
                .put("state")
                .put("zipcode")
                .put("country")
        );

        JSONArray jaddresses = new JSONArray();
        for (final Address address : addresses) {
            final JSONArray jaddress = new JSONArray();
            jaddress
                .put(address.getType())
                .put(address.isPreferred())
                .put(address.getPoBox())
                .put(address.getAddressLineOne())
                .put(address.getAddressLineTwo())
                .put(address.getCity())
                .put(address.getState())
                .put(address.getZipcode())
                .put(address.getCountry());

            jaddresses.put(jaddress);
        }
        jtable.put("values", jaddresses);
        return jtable;
    }

    public static JSONObject generateWeburlsTable(
        final String contactId, final WebURL[] weburls
    ) {
        JSONObject jtable = new JSONObject();
        jtable.put("caption", "Contact (" + contactId + ") Weburls:");
        jtable.put("headers", new JSONArray().put("type").put("address").put("preferred"));
        JSONArray jweburls = new JSONArray();
        for (final WebURL weburl : weburls) {
            final JSONArray jweburl = new JSONArray();
            jweburl
                .put(weburl.getType())
                .put(weburl.getUrl())
                .put(weburl.isPreferred());
            jweburls.put(jweburl);
        }
        jtable.put("values", jweburls);
        return jtable;
    }

    public static JSONObject generateQuickContactsTable(
        final QuickContact qcontact
    ) {
        JSONObject jtable = new JSONObject();
        jtable.put("caption", "Quick Contact:");
        jtable.put("headers", 
            new JSONArray()
                .put("contactId")
                .put("formattedName")
                .put("firstName")
                .put("middleName")
                .put("lastName")
                .put("prefix")
                .put("suffix")
                .put("nickName")
                .put("organization")
        );
        JSONArray jValuesEntry = new JSONArray()
            .put(qcontact.getContactId())
            .put(qcontact.getFormattedName())
            .put(qcontact.getFirstName())
            .put(qcontact.getMiddleName())
            .put(qcontact.getLastName())
            .put(qcontact.getPrefix())
            .put(qcontact.getSuffix())
            .put(qcontact.getNickname())
            .put(qcontact.getOrganization());

        jtable.put("values", new JSONArray().put(jValuesEntry));
        return jtable;
    }

    public static JSONObject generateContactsTable(final Contact contact) {
        JSONObject jtable = new JSONObject();
        jtable.put("caption", "Contact:");
        jtable.put("headers", 
            new JSONArray()
                .put("contactId")
                .put("creationDate")
                .put("modificationDate")
                .put("formattedName")
                .put("firstName")
                .put("lastName")
                .put("prefix")
                .put("suffix")
                .put("nickName")
                .put("organization")
                .put("jobTitle")
                .put("anniversary")
                .put("gender")
                .put("spouse")
                .put("hobby")
                .put("gender")
        );
        JSONArray jValuesEntry = new JSONArray()
            .put(contact.getContactId())
            .put(contact.getCreationDate())
            .put(contact.getModificationDate())
            .put(contact.getFormattedName())
            .put(contact.getFirstName())
            .put(contact.getLastName())
            .put(contact.getPrefix())
            .put(contact.getSuffix())
            .put(contact.getNickname())
            .put(contact.getOrganization())
            .put(contact.getJobTitle())
            .put(contact.getAnniversary())
            .put(contact.getGender())
            .put(contact.getSpouse())
            .put(contact.getHobby())
            .put(contact.getGender() == null ? null : contact.getGender().getString());

        jtable.put("values", new JSONArray().put(jValuesEntry));
        return jtable;
    }
}

/* vim: set expandtab tabstop=4 shiftwidth=4 softtabstop=4: */
